core/src/main/java/io/questdb/cairo/O3PartitionJob.java

```diff
@@ -1192,23 +1192,23 @@ private static void publishOpenColumnTasks(
             long tempIndexSize = mergeRowCount * TIMESTAMP_MERGE_ENTRY_BYTES;
             assert tempIndexSize > 0; // avoid SIGSEGV
 
-            if (!tableWriter.isDeduplicationEnabled()) {
-                timestampMergeIndexSize = tempIndexSize;
-                timestampMergeIndexAddr = createMergeIndex(
-                        srcTimestampAddr,
-                        sortedTimestampsAddr,
-                        mergeDataLo,
-                        mergeDataHi,
-                        mergeOOOLo,
-                        mergeOOOHi,
-                        timestampMergeIndexSize
-                );
-            } else {
-                final long tempIndexAddr = Unsafe.malloc(tempIndexSize, MemoryTag.NATIVE_O3);
-                final DedupColumnCommitAddresses dedupCommitAddresses = tableWriter.getDedupCommitAddresses();
-                final Path tempTablePath = Path.getThreadLocal(tableWriter.getConfiguration().getRoot()).concat(tableWriter.getTableToken());
+            try {
+                if (!tableWriter.isDeduplicationEnabled()) {
+                    timestampMergeIndexSize = tempIndexSize;
+                    timestampMergeIndexAddr = createMergeIndex(
+                            srcTimestampAddr,
+                            sortedTimestampsAddr,
+                            mergeDataLo,
+                            mergeDataHi,
+                            mergeOOOLo,
+                            mergeOOOHi,
+                            timestampMergeIndexSize
+                    );
+                } else {
+                    final long tempIndexAddr = Unsafe.malloc(tempIndexSize, MemoryTag.NATIVE_O3);
+                    final DedupColumnCommitAddresses dedupCommitAddresses = tableWriter.getDedupCommitAddresses();
+                    final Path tempTablePath = Path.getThreadLocal(tableWriter.getConfiguration().getRoot()).concat(tableWriter.getTableToken());
```

Comment:
there is a chance of leaking `tempIndexAddr` when this path allocation fails

Reply:
We made `NATIVE_PATH` allocation infallible, so this shouldn't ever fail.